Source/GTMSessionFetcher.m

```diff
@@ -3712,18 +3717,44 @@ - (nonnull dispatch_queue_t)callbackQueue {
   @synchronized(self) {
     GTMSessionMonitorSynchronized(self);
 
-    return _callbackQueue;
+    return _underlyingCallbackQueue;
   }  // @synchronized(self)
 }
 
 - (void)setCallbackQueue:(nullable dispatch_queue_t)queue {
   @synchronized(self) {
     GTMSessionMonitorSynchronized(self);
 
-    _callbackQueue = queue ?: dispatch_get_main_queue();
+    GTMSESSION_ASSERT_DEBUG(![self isFetchingUnsynchronized],
+                            @"callbackQueue should not change after beginFetch has been invoked");
+
+    _underlyingCallbackQueue = queue ?: dispatch_get_main_queue();
+    _callbackQueue = CreateCallbackQueueWithUnderlyingQueue(_underlyingCallbackQueue);
   }  // @synchronized(self)
 }
 
+static dispatch_queue_t CreateCallbackQueueWithUnderlyingQueue(
```

Comment:
Hmm I’d like to keep this logic separate from the getter logic to improve readability a bit since the getter logic has `@synchronized` stuff and lazy initialization stuff in it.